### DISCUSSION TITLE
Allow `<mark-message>` to fail gracefully if no messages to mark in list

### DISCUSSION THREAD

**Discussion by nickwynja** [neomutt/neomutt#4513] [Development]
As discussed in #3280, using `<mark-message>z<enter><change-folder>^<enter>'z` is a valid workaround to refresh a mailbox after changes were made in notmuch. This works nicely except for when the mailbox is empty.

`<mark-message>` fails since there is no message in the mailbox, which is fair. But this prevents the `<change-folder>` from happening, which is what refreshed the inbox.

## What would you like to be able to do?

I don't know if we're too far off track trying to solve the original problem, but having `<mark-message>` be able to fail gracefully if there are no messages and continue the rest of the operation of the macro would help solve my issue.

## What needs to change in NeoMutt?

As far as I can tell, there isn't any type of conditionals in macros so I couldn't configure to not run `<mark-message>` if the mailbox is empty.

Looks like `<mark-message>` runs the standard `prereq` check which makes sense in a normal case where you'd want it to fail when there are no messages in the list.

This might be introducing a dangerous new concept but something like `<mark-message?>` that can fail gracefully if there are no messages to mark and the macro can continue.

**Comment by flatcap** [comment]
Hmm... that **is** tricky.

Macros are very literal.
If you had a conditional `<mark-message?>` then what happens to the `z` that comes next?

You'd need a `<is-message-selected>[my macro]</is-message-selected>`
Or perhaps: `<cond:message>[my macro]<end-cond>`

Some similar ideas came up years ago #69
